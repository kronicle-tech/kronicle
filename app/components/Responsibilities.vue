<template>
  <!-- eslint-disable vue/require-v-for-key -->
  <ul v-if="responsibilities && responsibilities.length > 0">
    <li v-for="responsibility in responsibilities">
      <Markdown :markdown="responsibility.description" />
    </li>
  </ul>
  <!-- eslint-enable -->
</template>

<script lang="ts">
import Vue, { PropType } from 'vue'
import { Responsibility } from '~/types/component-catalog-service'
import Markdown from '~/components/Markdown.vue'

export default Vue.extend({
  components: {
    Markdown,
  },
  props: {
    responsibilities: {
      type: Array as PropType<Responsibility[]>,
      default: undefined,
    },
  },
})
</script>
