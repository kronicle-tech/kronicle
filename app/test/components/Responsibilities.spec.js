import { mount } from '@vue/test-utils'
import Responsibilities from '@/components/Responsibilities.vue'

describe('Responsibilities', () => {
  let propsData
  let wrapper
  const createWrapper = () => {
    wrapper = mount(Responsibilities, { propsData })
  }

  beforeEach(() => {
    propsData = {}
  })

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  describe('when responsibilities prop is not set', () => {
    test('renders nothing', () => {
      createWrapper()
      expect(wrapper.html()).toEqual('')
    })
  })

  describe('when responsibilities prop is set to an empty array', () => {
    beforeEach(() => {
      propsData.responsibilities = []
    })

    test('renders nothing', () => {
      createWrapper()
      expect(wrapper.html()).toEqual('')
    })
  })

  describe('when responsibilities prop is set to one responsibility', () => {
    beforeEach(() => {
      propsData.responsibilities = [
        {
          description: 'Test Description 1',
        },
      ]
    })

    test('renders an unordered list showing the responsibility', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when responsibilities prop is set to one responsibility with Markdown in the description', () => {
    beforeEach(() => {
      propsData.responsibilities = [
        {
          description: 'Text with *bold* formatting',
        },
      ]
    })

    test('renders the responsibility with the Markdown converted to HTML', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when responsibilities prop is set to multiple responsibilities', () => {
    beforeEach(() => {
      propsData.responsibilities = [
        {
          description: 'Test Description 1',
        },
        {
          description: 'Test Description 2',
        },
        {
          description: 'Test Description 3',
        },
      ]
    })

    test('renders an unordered list showing the responsibilities', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
