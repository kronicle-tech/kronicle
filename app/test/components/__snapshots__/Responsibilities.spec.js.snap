// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Responsibilities when responsibilities prop is set to multiple responsibilities renders an unordered list showing the responsibilities 1`] = `
"<ul>
  <li>
    <div>
      <p>Test Description 1</p>
    </div>
  </li>
  <li>
    <div>
      <p>Test Description 2</p>
    </div>
  </li>
  <li>
    <div>
      <p>Test Description 3</p>
    </div>
  </li>
</ul>"
`;

exports[`Responsibilities when responsibilities prop is set to one responsibility renders an unordered list showing the responsibility 1`] = `
"<ul>
  <li>
    <div>
      <p>Test Description 1</p>
    </div>
  </li>
</ul>"
`;

exports[`Responsibilities when responsibilities prop is set to one responsibility with Markdown in the description renders the responsibility with the Markdown converted to HTML 1`] = `
"<ul>
  <li>
    <div>
      <p>Text with <em>bold</em> formatting</p>
    </div>
  </li>
</ul>"
`;
